### ISSUE TITLE
Recursive datastructures result in segfault / hang

### ISSUE THREAD

**Issue by rainerschoe** [IBM/gWhisper#53]
caused by infinite depth grammar being created

Typical scenario:
```
./gwhisper localhost testService callWithRecursiveMessage
fish: “./gwhisper localhost teatSer…” terminated by signal SIGSEGV (Address boundary error)
```

**Comment by rainerschoe** [comment]
Nice example is also the
https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/struct.proto
struct datastructure which is recursive and causes endless recurion until stack overflows.

**Comment by rainerschoe** [comment]
As a quick-fix I will implement a limit of grammar nesting depth, which will essentially allow recursive grammars, as long as the users won't type too many levels of nesting on the commandline.

If we can use this as a permanent solution or not, we still need to decide, as it only allows "fake" recursion. I.e. all recursion gets serialized into one huge grammar tree.

**Comment by rainerschoe** [comment]
I think the 10-levels of nesting solution implemented in #93 is good enough to also be considered a permanent solution for this problem.
If someone really tries to pass more that 10 levels of nested messages to gWhisper commandline maybe gWhisper is not the right tool for the job :P
